<h1>Evaluation of Pre-Trained Models</h1>

<a id="item41"></a>

In [1]:
from tensorflow.keras.preprocessing.image import ImageDataGenerator
import tensorflow.keras
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Dense
#from tensorflow.keras.applications import VGG16
#from tensorflow.keras.applications.vgg16 import VGG16
from tensorflow.keras.applications.resnet50 import preprocess_input

In [2]:
model_resnet50 = tensorflow.keras.models.load_model('classifier_resnet_50_model.h5')

In [3]:
data_generator = ImageDataGenerator(
    preprocessing_function=preprocess_input,
)

In [4]:
image_resize = 224
num_classes = 2

In [5]:
test_generator = data_generator.flow_from_directory(
    'horse2zebra/testing',
    target_size=(image_resize, image_resize),
    shuffle=False)

Found 742 images belonging to 2 classes.


In [6]:
#scores = model.evaluate_generator(test_generator,len(test_generator)) 
scores = model_resnet50.evaluate_generator(test_generator, len(test_generator))
print("Loss = ", scores[0])
print("Accuracy = ", scores[1])
scores_resnet50 = scores

/home/alexandr/.local/lib/python3.6/site-packages/keras/engine/training.py:2006: UserWarning: `Model.evaluate_generator` is deprecated and will be removed in a future version. Please use `Model.evaluate`, which supports generators.
  warnings.warn('`Model.evaluate_generator` is deprecated and '


Loss =  0.02468941919505596
Accuracy =  0.9905660152435303


In [7]:
from tensorflow.keras.applications.resnet import preprocess_input
data_generator = ImageDataGenerator(
    preprocessing_function=preprocess_input,
)
image_resize = 224
num_classes = 2
test_generator = data_generator.flow_from_directory(
    'horse2zebra/testing',
    target_size=(image_resize, image_resize),
    shuffle=False)

Found 742 images belonging to 2 classes.


In [8]:
model_resnet101 = tensorflow.keras.models.load_model('classifier_resnet_101_model.h5')

In [9]:
scores = model_resnet101.evaluate_generator(test_generator,len(test_generator)) 
print("Loss = ", scores[0])
print("Accuracy = ", scores[1])
scores_resnet101 = scores

/home/alexandr/.local/lib/python3.6/site-packages/keras/engine/training.py:2006: UserWarning: `Model.evaluate_generator` is deprecated and will be removed in a future version. Please use `Model.evaluate`, which supports generators.
  warnings.warn('`Model.evaluate_generator` is deprecated and '


Loss =  0.022429844364523888
Accuracy =  0.993261456489563


In [10]:
from tensorflow.keras.applications.resnet import preprocess_input
data_generator = ImageDataGenerator(
    preprocessing_function=preprocess_input,
)
image_resize = 224
num_classes = 2
test_generator = data_generator.flow_from_directory(
    'horse2zebra/testing',
    target_size=(image_resize, image_resize),
    shuffle=False)

Found 742 images belonging to 2 classes.


In [11]:
model_resnet152 = tensorflow.keras.models.load_model('classifier_resnet_152_model.h5')

In [12]:
scores = model_resnet152.evaluate_generator(test_generator,len(test_generator)) 
print("Loss = ", scores[0])
print("Accuracy = ", scores[1])
scores_resnet152 = scores

Loss =  0.036423079669475555
Accuracy =  0.986522912979126


In [13]:
from tensorflow.keras.applications.vgg16 import preprocess_input
data_generator = ImageDataGenerator(
    preprocessing_function=preprocess_input,
)
image_resize = 224
num_classes = 2
test_generator = data_generator.flow_from_directory(
    'horse2zebra/testing',
    target_size=(image_resize, image_resize),
    shuffle=False)

Found 742 images belonging to 2 classes.


In [14]:
model_vgg16 = tensorflow.keras.models.load_model('classifier_vgg16_model.h5')

In [15]:
scores = model_vgg16.evaluate_generator(test_generator,len(test_generator)) 
print("Loss = ", scores[0])
print("Accuracy = ", scores[1])
scores_vgg16 = scores

Loss =  0.3364546000957489
Accuracy =  0.9218328595161438


In [16]:
from tensorflow.keras.applications.vgg19 import preprocess_input
data_generator = ImageDataGenerator(
    preprocessing_function=preprocess_input,
)
image_resize = 224
num_classes = 2
test_generator = data_generator.flow_from_directory(
    'horse2zebra/testing',
    target_size=(image_resize, image_resize),
    shuffle=False)

Found 742 images belonging to 2 classes.


In [17]:
model_vgg19 = tensorflow.keras.models.load_model('classifier_vgg19_model.h5')

In [18]:
scores = model_vgg19.evaluate_generator(test_generator,len(test_generator)) 
print("Loss = ", scores[0])
print("Accuracy = ", scores[1])
scores_vgg19 = scores

Loss =  0.28541600704193115
Accuracy =  0.9231805801391602


In [19]:
from tensorflow.keras.applications.inception_v3 import preprocess_input
data_generator = ImageDataGenerator(
    preprocessing_function=preprocess_input,
)
image_resize = 224
num_classes = 2
test_generator = data_generator.flow_from_directory(
    'horse2zebra/testing',
    target_size=(image_resize, image_resize),
    shuffle=False)

Found 742 images belonging to 2 classes.


In [20]:
model_inception_v3 = tensorflow.keras.models.load_model('classifier_inception_v3_model.h5')

In [21]:
scores = model_inception_v3.evaluate_generator(test_generator,len(test_generator)) 
print("Loss = ", scores[0])
print("Accuracy = ", scores[1])
scores_inception_v3 = scores

Loss =  0.03982914984226227
Accuracy =  0.9851751923561096


In [22]:
from tensorflow.keras.applications.inception_resnet_v2 import preprocess_input
data_generator = ImageDataGenerator(
    preprocessing_function=preprocess_input,
)
image_resize = 224
num_classes = 2
test_generator = data_generator.flow_from_directory(
    'horse2zebra/testing',
    target_size=(image_resize, image_resize),
    shuffle=False)

Found 742 images belonging to 2 classes.


In [23]:
model_inception_resnet_v2 = tensorflow.keras.models.load_model('classifier_inception_resnet_v2_model.h5')

In [24]:
scores = model_inception_resnet_v2.evaluate_generator(test_generator,len(test_generator)) 
print("Loss = ", scores[0])
print("Accuracy = ", scores[1])
scores_inception_resnet_v2 = scores

Loss =  0.016801562160253525
Accuracy =  0.993261456489563


In [25]:
loss = list()

loss.append(scores_resnet50[0])
loss.append(scores_resnet101[0])
loss.append(scores_resnet152[0])
loss.append(scores_vgg16[0])
loss.append(scores_vgg19[0])
loss.append(scores_inception_v3[0])
loss.append(scores_inception_resnet_v2[0])

accuracy = list()

accuracy.append(scores_resnet50[1])
accuracy.append(scores_resnet101[1])
accuracy.append(scores_resnet152[1])
accuracy.append(scores_vgg16[1])
accuracy.append(scores_vgg19[1])
accuracy.append(scores_inception_v3[1])
accuracy.append(scores_inception_resnet_v2[1])

labels = ['resnet50', 'resnet101', 'resnet152', 'vgg16', 'vgg19', 'inception_v3', 'inception_resnet_v2']

In [26]:
import pandas as pd
data = {'Model':labels, 'Loss':loss, 'Accuracy':accuracy}
test_df = pd.DataFrame(data)
test_df

Model      Loss  Accuracy
0             resnet50  0.024689  0.990566
1            resnet101  0.022430  0.993261
2            resnet152  0.036423  0.986523
3                vgg16  0.336455  0.921833
4                vgg19  0.285416  0.923181
5         inception_v3  0.039829  0.985175
6  inception_resnet_v2  0.016802  0.993261

| Model           	  | Loss 	 | Accuracy |
|---------------------|----------|----------|
| resnet50  	      | 0.024689 | 0.990566 |
| resnet101  	      | 0.022430 | 0.993261 |
| resnet152  	      | 0.036423 | 0.986523 |
| vgg16  	          | 0.336455 | 0.921833 |
| vgg19  	          | 0.285416 | 0.923181 |
| inception_v3  	  | 0.039829 | 0.985175 |
| inception_resnet_v2 |	0.016802 | 0.993261 |